### ISSUE TITLE
GStreamer Event Support on kvssink

### ISSUE THREAD

**Issue by suggestedfixes** [awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#375]
Currently, there isn't a way to send errors to the application layer. One possible approach is to support GStreamer events so that the application layer can handle errors.

**Comment by MushMal** [comment]
I agree that this is super important for any involved application using KVS GStreamer plugin.

To add a few notes on the events.

* Report any errors during streaming
* Report any error ACKs coming back
* Report any API call failures like putFrame/putMetadata, etc..
* Report most of the callbacks (informative or pressures) so the applications can react.
